plugins/webp-uploads/helper.php

```diff
@@ -240,8 +240,21 @@ function webp_uploads_generate_image_size( int $attachment_id, string $size, str
 		$size_data['height'] = $metadata['sizes'][ $size ]['height'];
 	}
 
-	if ( isset( $sizes[ $size ]['crop'] ) ) {
-		$size_data['crop'] = (bool) $sizes[ $size ]['crop'];
+	if (
+		isset( $sizes[ $size ]['crop'] ) &&
+		(
+			(
+				is_array( $sizes[ $size ]['crop'] ) &&
+				count( $sizes[ $size ]['crop'] ) === 2 &&
+				isset( $sizes[ $size ]['crop'][0] ) &&
+				isset( $sizes[ $size ]['crop'][1] ) &&
+				is_string( $sizes[ $size ]['crop'][0] ) &&
+				is_string( $sizes[ $size ]['crop'][1] )
+			) ||
+			is_bool( $sizes[ $size ]['crop'] )
+		)
+	) {
+		$size_data['crop'] = $sizes[ $size ]['crop'];
```

Comment:
I don't think we need to do all this granular validation on the `crop` key. WordPress Core doesn't do it and lets everything through. It only ensures at the very last point when it's actually used (in the `image_resize_dimensions()` function) that it has the correct format.

I think allowing any value here to pass through makes more sense as it makes our code aligned with Core and more future-proof, in case Core would later add support for other types for the `crop` key. It's also in line with `width` and `height` above, where we don't verify the type.
```suggestion
	if ( isset( $sizes[ $size ]['crop'] ) ) {
		$size_data['crop'] = $sizes[ $size ]['crop'];
```

Reply:
I added granular checks to ensure that the Modern Image Formats code does not produce errors on invalid input. However, to future-proof code, as you suggested, simplifying the condition makes more sense.